From c68ded695938b43682d4bd7dfaf40e5b267dfe3b Mon Sep 17 00:00:00 2001
From: Yoann Desmouceaux <ydesmouc@cisco.com>
Date: Fri, 19 Feb 2016 12:49:29 +0100
Subject: [PATCH] enic: fix dma addr of outgoing packets

The enic PMD driver send function uses a constant offset instead of relying on the data_off in the mbuf to find the start of the packet.

Signed-off-by: Yoann Desmouceaux <ydesmouc@cisco.com>
---
 drivers/net/enic/enic_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/enic/enic_main.c b/drivers/net/enic/enic_main.c
index 07a9810..f818c32 100644
--- a/drivers/net/enic/enic_main.c
+++ b/drivers/net/enic/enic_main.c
@@ -166,7 +166,7 @@ void enic_send_pkt(struct enic *enic, struct vnic_wq *wq,
 	uint16_t mss = 0;
 	uint8_t vlan_tag_insert = 0;
 	uint64_t bus_addr = (dma_addr_t)
-	    (tx_pkt->buf_physaddr + RTE_PKTMBUF_HEADROOM);
+	    (tx_pkt->buf_physaddr + tx_pkt->data_off);
 
 	if (sop) {
 		if (ol_flags & PKT_TX_VLAN_PKT)
-- 
2.1.4

